plugins/basestate/BaseState.cc

```diff
@@ -0,0 +1,243 @@
+#include "BaseStateDriver.cpp"
+#include <gz/plugin/Register.hh>
+#include <gz/sim/Joint.hh>
+#include <gz/sim/Link.hh>
+#include <gz/sim/Model.hh>
+#include <gz/sim/Sensor.hh>
+#include <gz/sim/System.hh>
+#include <gz/sim/Util.hh>
+#include <gz/sim/components/Sensor.hh>
+#include <gz/transport/Node.hh>
+#include <iostream>
+#include <yarp/dev/PolyDriver.h>
+#include <yarp/dev/PolyDriverList.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/Network.h>
+
+using namespace gz;
+using namespace sim;
+using namespace systems;
+
+namespace gzyarp
+{
+
+class BaseState : public System, public ISystemConfigure, public ISystemPostUpdate
+{
+public:
+    BaseState()
+        : m_deviceRegistered(false)
+    {
+    }
+
+    virtual ~BaseState()
+    {
+        if (m_deviceRegistered)
+        {
+            Handler::getHandler()->removeDevice(m_deviceScopedName);
+            m_deviceRegistered = false;
+        }
+
+        if (m_baseStateDriver.isValid())
+            m_baseStateDriver.close();
+        BaseStateDataSingleton::getBaseStateDataHandler()->removeBaseLink(m_baseLinkScopedName);
+        yarp::os::Network::fini();
+    }
+
+    virtual void Configure(const Entity& _entity,
+                           const std::shared_ptr<const sdf::Element>& _sdf,
+                           EntityComponentManager& _ecm,
+                           EventManager& /*_eventMgr*/) override
+    {
+        yarp::os::Network::init();
+        if (!yarp::os::Network::checkNetwork())
+        {
+            yError() << "Yarp network does not seem to be available, is the yarpserver running?";
+            return;
+        }
```

Comment:
```suggestion
```

Again, also this is everyhere, so it is not your fault. However, this comes from another era, in which all network wrapper servers were using the YARP network. As now wrapper could just use other stuff (like ROS2), we can avoid to check if network is available. Network wrapper servers that use YARP ports will care if a YARP network is actually available.